server/src/lib/challenge.ts

```diff
@@ -31,74 +28,82 @@ export default class Challenge {
   }
 
   getPoints = async (
-    { client_id, params: { challeng_token } }: Request,
+    { client_id, params: { challenge } }: Request,
     response: Response
   ) => {
-    console.log(
-      `[DEBUG] Challenge.getPoints() - client_id:${client_id}, challenge_token:${challeng_token}`
-    );
-    let data = getMockData();
-
-    response.json(data.challengePoint);
+    response.json(await this.model.db.getUserPoints(client_id, challenge));
   };
 
   getWeeklyProgress = async (
-    { client_id, params: { challeng_token } }: Request,
+    { client_id, params: { challenge } }: Request,
     response: Response
   ) => {
-    console.log(
-      `[DEBUG] Challenge.getWeeklyProgress - client_id:${client_id}, challenge_token:${challeng_token}`
+    const progress = await this.model.db.getWeeklyProgress(
+      client_id,
+      challenge
     );
-    let data = getMockData();
-
-    response.json(data.weeklyChallenge);
+    const weeklyProgress = {
+      week: progress.week || null,
```

Comment:
Do we need these `|| null`s here? In what case will these values be falsey?

Reply:
good point, there shouldn't be these `|| null` here